pkg/bpf/bpf_kmesh_workload.go

```diff
@@ -458,17 +465,175 @@ func (sm *BpfSendMsgWorkload) Detach() error {
 	return nil
 }
 
+type podLinkInfo struct {
+	// netNsPath is the net namespace path of this pod, starts with /host/proc
+	netNsPath string
+	// linkedIfNames is a set of NIC interfaces names with XDP program linked on
+	linkedIfNames sets.Set[string]
+}
+
 type BpfXdpAuthWorkload struct {
 	Info BpfInfo
 	Link link.Link
 	bpf2go.KmeshXDPAuthObjects
+	// podLinkMap contains entries which key is pod UID, and value is XDP program link information on this pod
+	podLinkMap map[types.UID]podLinkInfo
+	// rwLock R/W lock for podLinkMap
+	rwLock sync.RWMutex
+}
+
+// AddPodLink links XDP program on all non-loopback, up NIC interfaces except loopback in the pod which UID is podUID,
+// once successful, adds a new {podUID, podLinkInfo} into podLinkMap
+func (xa *BpfXdpAuthWorkload) AddPodLink(podUID types.UID, netNsPath string) error {
+	// If pod already exists in the map, just return
+	xa.rwLock.RLock()
+	if _, ok := xa.podLinkMap[podUID]; ok {
+		xa.rwLock.RUnlock()
+		return nil
+	}
+	xa.rwLock.RUnlock()
+
+	podLnInf := podLinkInfo{
+		netNsPath:     netNsPath,
+		linkedIfNames: sets.New[string](),
+	}
+	if err := netns.WithNetNSPath(netNsPath, func(netns.NetNS) error {
+		// Get all NIC iface in a pod
+		ifaces, err := net.Interfaces()
+		if err != nil {
+			return err
+		}
+		log.Infof("Interfaces in pod: %v is: %v", podUID, ifaces)
+
+		// Link XDP prog on every iface, except loopback or not up
+		for _, iface := range ifaces {
+			if iface.Flags&net.FlagLoopback != 0 || iface.Flags&net.FlagUp == 0 {
```

Comment:
should we exclude linklocal unicast?

Reply:
seems there is no corresponding flags in package `net`